src/fu-util.c

```diff
@@ -1043,6 +1064,23 @@ fu_util_device_test_filename(FuUtilPrivate *priv,
 		json_builder_set_member_name(helper->builder, "interactive");
 		json_builder_add_boolean_value(helper->builder, interactive);
 	}
+	if (json_object_has_member(json_obj, "cpu-architectures")) {
+		JsonArray *json_array = json_object_get_array_member(json_obj, "cpu-architectures");
+		gboolean matched = FALSE;
+		const gchar *arch = g_hash_table_lookup(helper->report_metadata, "CpuArchitecture");
+		for (guint i = 0; i < json_array_get_length(json_array); i++) {
+			const gchar *arch_tmp = json_array_get_string_element(json_array, i);
+			if (g_strcmp0(arch, arch_tmp) == 0) {
+				matched = TRUE;
+				break;
+			}
+		}
+		if (!matched) {
+			g_debug("skipping as wrong architecture [%s]", arch);
+			helper->nr_success++;
+			return TRUE;
+		}
```

Comment:
Shouldn't this be a false and nothing to do?

Reply:
How about half way between - can we at least show a warning?

Reply:
Good idea, done.